stage1_fly/run/main.go

```diff
@@ -0,0 +1,368 @@
+	for _, tuple := range namedVolumeMounts {
+		if _, isHostMount := hostMounts[tuple.V.Source]; isHostMount {
+			// Mark the host mount as SHARED so the container's changes to the mount are propagated to the host
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", "", tuple.V.Source, "none", syscall.MS_REC | syscall.MS_SHARED},
+			)
+		}
+		argFlyMounts = append(argFlyMounts,
+			flyMount{tuple.V.Source, rfs, tuple.M.Path, "none", flags},
+		)
+
+		if tuple.V.ReadOnly != nil && *tuple.V.ReadOnly {
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", rfs, tuple.M.Path, "none", flags | syscall.MS_REMOUNT | syscall.MS_RDONLY},
+			)
+		}
+	}
+	return argFlyMounts, nil
+}
+
+func stage1() int {
+	uuid, err := types.NewUUID(flag.Arg(0))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "UUID is missing or malformed\n")
+		return 1
+	}
+
+	root := "."
+	p, err := stage1commontypes.LoadPod(root, uuid)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "can't load pod: %v\n", err)
+		return 1
+	}
+
+	if len(p.Manifest.Apps) != 1 {
+		fmt.Fprintf(os.Stderr, "flavor %q only supports 1 application per Pod for now.\n", flavor)
+		return 1
+	}
+
+	lfd, err := common.GetRktLockFD()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "can't get rkt lock fd: %v\n", err)
+		return 1
+	}
+
+	// set close-on-exec flag on RKT_LOCK_FD so it gets correctly closed when invoking network plugins
+	if err := sys.CloseOnExec(lfd, true); err != nil {
+		fmt.Fprintf(os.Stderr, "can't set FD_CLOEXEC on rkt lock: %v\n", err)
+		return 1
+	}
+
+	// TODO: insert environment from manifest
+	env := []string{"PATH=/bin:/sbin:/usr/bin:/usr/local/bin"}
+	args := p.Manifest.Apps[0].App.Exec
+	rfs := filepath.Join(common.AppPath(p.Root, p.Manifest.Apps[0].Name), "rootfs")
+
+	argFlyMounts, err := evaluateMounts(rfs, string(p.Manifest.Apps[0].Name), p)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "can't evaluate mounts: %v\n", err)
+		return 1
+	}
+
+	effectiveMounts := append(
+		[]flyMount{
+			{"", "", "/dev", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/dev", rfs, "/dev", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/proc", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/proc", rfs, "/proc", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/sys", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/sys", rfs, "/sys", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"tmpfs", rfs, "/tmp", "tmpfs", 0},
+		},
+		argFlyMounts...,
+	)
+
+	for _, mount := range effectiveMounts {
+		var (
+			err            error
+			hostPathInfo   os.FileInfo
+			targetPathInfo os.FileInfo
+		)
+
+		if strings.HasPrefix(mount.HostPath, "/") {
+			if hostPathInfo, err = os.Stat(mount.HostPath); err != nil {
+				fmt.Fprintf(os.Stderr, "stat of host directory %s: \n%v", mount.HostPath, err)
+				return 1
+			}
+		} else {
+			hostPathInfo = nil
+		}
+
+		absTargetPath := filepath.Join(mount.TargetPrefixPath, mount.RelTargetPath)
+		if absTargetPath != "/" {
```

Comment:
If I'm not mistaken, it can't anymore. I originally had a bind-mount for "/" to make it appear in the mount table when overlayfs is disabled. The if-clause should be safe to remove now.